mindmeld/cli.py

```diff
@@ -85,6 +98,133 @@ def _app_cli(ctx):
+)
+@click.option(
+    "--save", is_flag=True, required=False, help="Save built models using dvc"
+)
+@click.option("--checkout", required=False, help="Instantiate DVC within a repository")
+def dvc(ctx, init, save, checkout):
+    app = ctx.obj.get("app")
+    app_path = app.app_path
+
+    # Ensure that DVC is installed
+    if not which("dvc"):
+        logger.error(
+            "DVC is not installed. You can install DVC by running 'pip install dvc'"
+        )
+        return
+
+    if init:
+        success, error_string = _bash_helper(["dvc", "init"])
+        if not success:
+            logger.error("Error during initialization: %s", error_string)
+            return
+
+        # Set up a local remote
+        local_remote_path = get_dvc_local_remote_path(app_path)
+        success, error_string = _bash_helper(
+            ["dvc", "remote", "add", "-d", "myremote", local_remote_path]
+        )
+        if not success:
+            logger.error("Error during local remote set up: %s", error_string)
+            return
+
+        # Add DVC config file to staging
+        success, error_string = _bash_helper(["git", "add", ".dvc/config"])
+        if not success:
+            logger.error("Error while adding dvc config file: %s", error_string)
+            return
+
+        logger.info(
+            "Instantiated DVC repo and set up local remote in %s", local_remote_path
+        )
+        logger.info(
+            "The newly generated dvc config file (.dvc/config) has been added to git staging"
+        )
+    elif save:
+        generated_model_folder = get_generated_data_folder(app_path)
+
+        success, error_string = _dvc_add_helper(generated_model_folder)
+        if not success:
+            logger.error("Error during saving: %s", error_string)
+            return
+
+        success, error_string = _bash_helper(["dvc", "push"])
+        if not success:
+            logger.error("Error during dvc push: %s", error_string)
+            return
+
+        success, error_string = _bash_helper(
+            ["git", "add", "{}/.generated.dvc".format(app_path)]
+        )
+        if not success:
+            logger.error("Error adding model dvc file: %s", error_string)
+            return
+
+        logger.info("Successfully added .generated model folder to dvc")
+        logger.info(
+            "The newly generated .dvc file (%s/.generated.dvc) has been added to git staging",
+            app_path,
+        )
+    elif checkout:
+        success, error_string = _bash_helper(["git", "checkout", checkout])
+        if not success:
+            logger.error("Error during git checkout: %s", error_string)
+            return
+
+        success, error_string = _bash_helper(["dvc", "pull"])
+        if not success:
+            logger.error("Error during dvc checkout: %s", error_string)
+            return
+
+        logger.info(
+            "Successfully checked out models corresponding to hash %s", checkout
+        )
```

Comment:
In the else case we can print a helpful message (like the help option) like `python -m [app] dvc -h`

```
Options:
  --init           Instantiate DVC within a repository
  --save           Save built models using dvc
  --checkout TEXT  Instantiate DVC within a repository
  -h, --help       Show this message and exit.

Reply:
If there is an error, it won't reach this point and it will print the error message that occurred with an additional help message. I'll add the help flag option per a comment below